openst: fix error 409 multiple networks

m-ildefons opened the pull request:
Specify public network.

This fixes the error 409 "multiple possible networks":
```
    + jcs --os-network-public Ext-Net create --cloud openstack --instance-type s1-8 --jenkins-credential storage-automation-for-root-user minimal-opensuse-15.4-x86_64 sesdev-integration_master-19
    Connected to jenkins server 2.378
    Traceback (most recent call last):
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/bin/jcs", line 8, in <module>
        sys.exit(main())
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/jcs/__init__.py", line 289, in main
        args.func(args)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/jcs/__init__.py", line 189, in _do_create
        args.os_network_fixed, args.os_network_public, args.os_security_groups)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/jcs/openst.py", line 45, in os_instance_create
        wait=True, auto_ip=True)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/decorator.py", line 232, in fun
        return caller(func, *(extras + args), **kw)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/openstack/cloud/_utils.py", line 235, in func_wrapper
        return func(*args, **kwargs)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/openstack/cloud/_compute.py", line 922, in create_server
        server = self.compute.create_server(**kwargs)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/openstack/compute/v2/_proxy.py", line 652, in create_server
        return self._create(_server.Server, **attrs)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/openstack/proxy.py", line 581, in _create
        return res.create(self, base_path=base_path)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/openstack/resource.py", line 1487, in create
        self._translate_response(response, has_body=has_body)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/openstack/resource.py", line 1254, in _translate_response
        exceptions.raise_from_response(response, error_message=error_message)
      File "/tmp/ws/sesdev-integration/master/19/jcs/venv/lib/python3.6/site-packages/openstack/exceptions.py", line 235, in raise_from_response
        http_status=http_status, request_id=request_id
    openstack.exceptions.ConflictException: ConflictException: 409: Client Error for url: https://compute.de1.cloud.ovh.net/v2.1/68bb3d2bb759469c94d53c072b024f22/servers, Multiple possible networks found, use a Network ID to be more specific.
    script returned exit code 1
```

Signed-off-by: Moritz Röhrich <moritz.rohrich@suse.com>

m-ildefons committed:
openst: fix error 409 multiple networks

Specify public network.

Signed-off-by: Moritz Röhrich <moritz.rohrich@suse.com>

kshtsk commented on jcs/openst.py:
```diff
@@ -41,7 +41,7 @@ def os_instance_create(self, instance_name, instance_type, image_name, key_name,
         instance = self._conn.create_server(
             instance_name, image=image, flavor=instance_type, key_name=key_name,
-            network=network_fixed, security_groups=security_groups,
+            network=network_public, security_groups=security_groups,
```
I suppose the reason it was working before, because while it was trying to the fixed network in the line 39 it was return Null value and passed as a value for parameter _network_, and because there was just one public network it was just taking it. Now there is more than one network on some region, so it started to fail.
This change is fine, but it make useless the line 39, and the parameter *network_fixed* in this method. So I suggest you to take the fixed network into account, for example, providing parameter like `network=network_fixed or network_public`

m-ildefons replied:
Makes sense. Using `network_public` only in case `network_fixed` is a `null` value is indeed less risky wrt. backwards compatibility.

m-ildefons committed:
fix: handle 'fixed' network

Use the public network in place of the fixed network only if the fixed
network couldn't be found. This doesn't brake backwards compatibility.

Signed-off-by: Moritz Röhrich <moritz.rohrich@suse.com>

kshtsk commented on jcs/openst.py:
```diff
@@ -36,12 +36,12 @@ def os_instance_create(self, instance_name, instance_type, image_name, key_name,
             raise Exception('OS instance type (flavor) "{}" not found'.format(instance_type))
 
-        network_fixed = self._conn.get_network(network_fixed)
+        network_fixed = self._conn.get_network(network_fixed) or network_public
```
Hm... this is kinda mix of value types, while network_fixed is lookup, the network_public remains just a string. I guess create_server can deal with that. But maybe it is better to lookup the public network also and report user if none of the network are found?

kshtsk replied:
ah.... and there is another thing.... look later how public network is used for making a floating ip. This code mostly used for our internal openstack cloud where we have floating ips, while OVH does not provide floating IP mechanism.
